src/views/Exam/Exam.utils.ts

```diff
@@ -11,13 +11,26 @@ export function getObjectValue<T>(
 export const getRandomQuestions = (
   questions: Question[],
   questionsCount: number,
+  learntQuestions?: LearntQuestions,
 ) => {
-  return shuffle(questions, { copy: true }).slice(0, questionsCount);
+  return shuffle(
+    learntQuestions?.length
+      ? questions.filter(({ id }) => !learntQuestions.includes(id))
+      : questions,
```

Comment:
Let's be a bit more explicit
```suggestion
    learntQuestions !== undefined && learntQuestions.length > 0
      ? questions.filter(({ id }) => !learntQuestions.includes(id))
      : questions,
```